y2020/ex22/src/lib.rs

```diff
@@ -36,7 +47,7 @@ impl FromStr for Deck {
     }
 }
 
-type GameHistory = HashSet<(Deck, Deck)>;
+type GameHistory = HashSet<(usize, usize)>;
```

Comment:
It would be interesting to see what happens if we switch this `HashSet` with a [`BTreeSet`](https://doc.rust-lang.org/stable/std/collections/struct.BTreeSet.html)... In theory I expect it to be slower, but given that we will probably have a small enough cardinality it might actually go faster!

Reply:
Then let's leave hashset as it seems more "intuitive" (at least to me...)